src/platform_impl/windows/event_loop.rs

```diff
@@ -245,17 +257,13 @@ impl EventLoop {
         }
 
         let exit_code = loop {
-            self.wait_and_dispatch_message(None);
-
-            if let Some(code) = self.exit_code() {
-                break code;
-            }
-
             self.dispatch_peeked_messages();
 
             if let Some(code) = self.exit_code() {
                 break code;
             }
+
+            self.wait_for_messages(None);
```

Comment:
Please clarify why we've removed the `exit_code()` check between waiting for messages and dispatching peeked messages.

Reply:
As I understand, it was here to handle cases when exit codes were set in callbacks to messages that called in `wait_and_dispatch_message`. I don't do any dispatch in new `self.wait_for_messages` so there is no new exit code that can appear before `dispatch_peeked_messages`. Also, before the loop, exit code is reset at https://github.com/rust-windowing/winit/blob/c858ac7d157535250a2ea35deb70b9e687bd4a86/src/platform_impl/windows/event_loop.rs#L231 so it cannot be set before first iteration of loop either.

Reply:
It seems notable that the ordering of `dispatch_peeked_` followed by `wait_for_messages` (instead of waiting then dispatching) here isn't consistent with the `pump_app_events` implementation, which seems like it could affect the order of dispatching some initial events for an app.

I believe `wait_for_messages` dispatches the `AboutToWait` event (`runner.prepare_wait()`) to notify the app that it's about to block and wait for input so I wonder if it might still make sense to check for a possible exit code from the app after `wait_for_messages()`